src/utils/config_utils/src/config_utils.c

```diff
@@ -330,9 +389,126 @@ bool ADUC_ConfigInfo_Init(ADUC_ConfigInfo* config, const char* configFilePath)
         goto done;
     }
 
+    const char* aduShellFolder = ADUC_JSON_GetStringFieldPtr(root_value, CONFIG_ADU_SHELL_FOLDER);
+
+    if (aduShellFolder != NULL)
+    {
+        if (mallocAndStrcpy_s(&(config->aduShellFolder), aduShellFolder) != 0)
+        {
+            goto done;
+        }
+    }
+    else
+    {
+        if (mallocAndStrcpy_s(&(config->aduShellFolder), ADUSHELL_FOLDER) != 0)
+        {
+            goto done;
+        }
+    }
+
+    config->aduShellFilePath = ADUC_StringFormat("%s/%s", config->aduShellFolder, ADUSHELL_FILENAME);
+    if (config->aduShellFilePath == NULL)
+    {
+        Log_Error("Failed to allocate memory for adu-shell file path");
+        goto done;
+    }
+
+    const char* dataFolder = ADUC_JSON_GetStringFieldPtr(root_value, CONFIG_ADU_DATA_FOLDER);
+
+    if (dataFolder != NULL)
+    {
+        if (mallocAndStrcpy_s(&(config->dataFolder), dataFolder) != 0)
+        {
+            goto done;
+        }
+    }
+    else
+    {
+        if (mallocAndStrcpy_s(&(config->dataFolder), ADUC_DATA_FOLDER) != 0)
+        {
+            goto done;
+        }
+    }
+
+    const char* downloadsFolder = ADUC_JSON_GetStringFieldPtr(root_value, CONFIG_ADU_DOWNLOADS_FOLDER);
+
+    if (downloadsFolder != NULL)
+    {
+        if (mallocAndStrcpy_s(&(config->downloadsFolder), downloadsFolder) != 0)
+        {
+            goto done;
+        }
+    }
+    else
+    {
+        config->downloadsFolder = ADUC_StringFormat("%s/%s", config->dataFolder, "downloads");
+        if (config->downloadsFolder == NULL)
+        {
+            Log_Error("Failed to allocate memory for downloads folder");
+            goto done;
+        }
+    }
+
+    const char* extensionsFolder = ADUC_JSON_GetStringFieldPtr(root_value, CONFIG_ADU_EXTENSIONS_FOLDER);
+
+    if (extensionsFolder != NULL)
+    {
+        if (mallocAndStrcpy_s(&(config->extensionsFolder), extensionsFolder) != 0)
+        {
+            goto done;
+        }
+    }
+    else
+    {
+        config->extensionsFolder = ADUC_StringFormat("%s/%s", config->dataFolder, "extensions");
+        if (config->extensionsFolder == NULL)
+        {
+            Log_Error("Failed to allocate memory for extension folder");
+            goto done;
+        }
+    }
+
+    if (NULL
+        == (config->extensionsComponentEnumeratorFolder =
+                ADUC_StringFormat("%s/%s", config->extensionsFolder, ADUC_EXTENSIONS_SUBDIR_COMPONENT_ENUMERATOR)))
+    {
+        Log_Error("Failed to allocate memory for component enumerator extension folder");
+        goto done;
+    }
+
+    if (NULL
```

Comment:
Seems like it needs clang-format?

Reply:
That's what caused by running clang-format....